packages/context/src/resolution-session.ts

```diff
@@ -369,46 +382,26 @@ export interface ResolutionContext<T = unknown> {
  * Error for context binding resolutions and dependency injections
  */
 export class ResolutionError extends Error {
-  constructor(
-    message: string,
-    readonly resolutionCtx: Partial<ResolutionContext>,
-  ) {
-    super(ResolutionError.buildMessage(message, resolutionCtx));
+  constructor(reason: string, readonly ctx: Partial<ResolutionContext>) {
+    super(`${reason} (${ResolutionError.buildMessage(ctx)})`);
     this.name = ResolutionError.name;
   }
 
-  private static buildDetails(resolutionCtx: Partial<ResolutionContext>) {
-    return {
-      context: resolutionCtx.context?.name ?? '',
-      binding: resolutionCtx.binding?.key ?? '',
-      resolutionPath: resolutionCtx.options?.session?.getResolutionPath() ?? '',
-    };
-  }
-
   /**
    * Build the error message for the resolution to include more contextual data
    * @param reason - Cause of the error
-   * @param resolutionCtx - Resolution context
+   * @param ctx - Resolution context
    */
-  private static buildMessage(
-    reason: string,
```

Comment:
Was the `reason` parameter removed intentionally?

Otherwise, this might constitute a breaking change as the class isn't sealed.

Reply:
It is not parameter being removed, but the whole private method.